addons/sale/views/sale_views.xml

```diff
@@ -439,7 +564,14 @@
                     <field name="user_id"/>
                     <field name="team_id" string="Sales Channel"/>
                     <field name="analytic_account_id"/>
-                    <field name="product_id"/>
+                    <!-- We only allow to search on the following sale order line fields (product, name) because the other fields, such as price, quantity, ...
+                        will not be searched as often, and if they need to be searched it's usually in the context of products
+                        and then they can be searched from the page listing the sale order lines related to a product (from the product itself).
+                    -->
+                    <field name="order_line" string="Line Product" filter_domain="[('order_line.product_id.name', 'ilike', self)]"/>
+                    <field name="order_line" string="Line Description" filter_domain="[('order_line.name', 'ilike', self), ('order_line.line_type', '=', 'line_accountable'), ]"/>
```

Comment:
Do we really need to be able to search on line description, notes, section, etc? Look at the resulting autocompletion window, for instance...